### DISCUSSION TITLE
Creating collections from collections in eleventy.js

### DISCUSSION THREAD

**Discussion by lucasnantonio** [11ty/eleventy#2403] [Q&A]
I'm trying to create a collection `colA` in eleventy.js, then access it when creating a second collection `colB`.

This has been showing flaky behavior, the build breaks 50% of the time because `colA` is undefined. 

Could this be because my first function is still running when the second one tries to access the first collection? If so, what is the best way to wait for the first one to complete before creating the second collection?

What else am I missing?

```
  eleventyConfig.addCollection("colA", function (collectionApi) {
    ....
    return colA;
  });

  eleventyConfig.addCollection("colB", function (collectionApi) {
  let colA = collectionApi.items[0].data.collections.colA;
    ....
    return colB;
  });
```

**Comment by pdehaan** [comment]
Since I don't think collection creation is synchronous or in a predictable order, I'm not sure there is a good solution, apart from possibly recreating `colA` within `colB`.
Or in some cases (like creating a tag list from an existing collection), you can often use a custom filter instead of a collection.

**Reply by lucasnantonio** [reply]
Yeah I couldn't find any info on whether collection creation is synchronous in the docs. Thanks for the suggestion about using filters instead. 

I'm wondering now if I shouldn't just export colA and colB as [global data from a .js file](https://www.11ty.dev/docs/data-js/) where I can make all my transformations. 

 I don't get the difference between doing this vs. creating collections or filters in the eleventy.js file though.
